### PR TITLE
Integrate linter for shell scripts

### PR DESCRIPTION
**Pull request by semeniak97mf**
Added the installation of the shellcheck linter in the pipeline. Fixed all the available warnings in existing scripts after linting. 
(The last PR was in my fork. Now in a separate branch in this repo)

https://github.com/Atleta-network/atleta/pull/139

### CODE REVIEW

File: deploy/testnet/run-validator.sh [hunk 1]
```diff
@@ -76,21 +76,19 @@ wait_availability() {
...
 
     while [ $retry_count -lt $max_retries ]; do
-        # Use curl to test the connection without making an actual request
-        curl --connect-timeout 5 "$rpc_api_endpoint" 2>/dev/null
-        
-        # Check the exit status of curl
-        if [ $? -eq 0 ]; then
+
+        # Use curl to test the connection without making an actual request and Check the exit status of curl
+        if  curl --connect-timeout 5 "$rpc_api_endpoint" 2>/dev/null; then
```
Comment (hrls) [inline deploy/testnet/run-validator.sh]: Space dup after `if`. `s/\s\s/\s`

File: deploy/testnet/revert.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ source ./config.env
 
 container_name="honest_worker"
 chainspec="./chainspec.json"
-rpc_api_endpoint="http://127.0.0.1:9944"
+#rpc_api_endpoint="http://127.0.0.1:9944"
```
Comment (hrls) [inline deploy/testnet/revert.sh]: It looks like this variable is being used here. I remember there was an unused variable error and it seems strange.
_Update_: my bad, used in another script. Just delete it or pass to another scripts as an argument.

**Review by hrls — DISMISSED** [review]

File: deploy/testnet/revert.sh (hunk 2 above)
Comment (semeniak97mf) [reply deploy/testnet/revert.sh]: Yes, it sends a warning that it is not being used.
```
In deploy/testnet/revert.sh line 19:
rpc_api_endpoint="http://127.0.0.1:9944"
^--------------^ SC2034 (warning): rpc_api_endpoint appears unused. Verify use (or export if used externally).
```
Maybe this line should be uncommented? And add a command, for example `echo "$rpc_api_endpoint"`. This command will be used as a stub for the linter and will not trigger the warning

File: .github/workflows/check.yml [hunk 3]
```diff
@@ -41,3 +41,5 @@ jobs:
           | jq '.packages | .[] | .name' \
           | rg --invert-match 'atleta-(node|runtime)' \
           | xargs -I {} cargo clippy --package {} --all-targets --all-features -- --deny warnings --deny missing_docs
+    - name: Run shellcheck
+      run: fdfind --type executable | xargs -I {} shellcheck {}
```
Comment (hrls) [inline .github/workflows/check.yml]: JFYI `fd-find` has a `--exec` option which behave the same as `xargs`, but with some additional tweaks of substitution of matched file path.

Comment (semeniak97mf) [reply .github/workflows/check.yml]: Ok, this command looks simpler. 
`fdfind --type executable --exec shellcheck {}`
Tested, it works. Thanks for the comment

File: deploy/testnet/revert.sh (hunk 2 above)
Comment (hrls) [reply deploy/testnet/revert.sh]: Just ensure this script do not call another one where this variable is expected to be set, and then remove this line entirely. Also you can add `set -u` to the scripts to fail any attempts to use undefined variables.

Comment (semeniak97mf) [reply deploy/testnet/revert.sh]: Done

File: deploy/testnet/run-validator.sh (hunk 1 above)
Comment (semeniak97mf) [reply deploy/testnet/run-validator.sh]: Thanks, fixed

**Review by hrls — DISMISSED** [review]

**Review by ales-tsurko — DISMISSED** [review]

**Review by hrls — APPROVED** [review]

**Pull request merged**